src/main/java/anto/TaskList.java

```diff
@@ -0,0 +1,218 @@
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    /**
+     * Creates a new TaskList class.
+     *
+     * @param taskList Initial ArrayList of tasks.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Add task given to arraylist.
+     *
+     * @param newTask Task to be added.
+     * @throws AntoException Throws anto exception if there is an IO Exception
+     */
+    public void addToList(Task newTask) throws AntoException {
+        this.taskList.add(newTask);
+
+        String relativePath = "data/anto.txt";
+        Path absolutePath = Paths.get(relativePath).toAbsolutePath();
+        File antoFile = absolutePath.toFile();
+
+        try {
+            FileWriter writer = new FileWriter(antoFile, true);
+            if (newTask instanceof Todo) {
+                writer.write(String.format("\nT | %s | %s", newTask.isDone ? "1" : "0", newTask.description));
+            } else if (newTask instanceof Deadline) {
+                writer.write(String.format("\nD | %s | %s | %s", newTask.isDone ? "1" : "0", newTask.description, (
+                        (Deadline) newTask).by.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"))));
+            } else if (newTask instanceof Event) {
+                writer.write(String.format("\nE | %s | %s | %s | %s", newTask.isDone ? "1" : "0", newTask.description, (
+                        (Event) newTask).from.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm")), (
+                        (Event) newTask).to.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"))));
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new AntoException("OOPS!!! IOException");
+        }
+    }
+
+    /**
+     * Returns current array list of tasks.
+     *
+     * @return ArrayList of tasks.
+     */
+    public ArrayList<Task> getStorage() {
+        return this.taskList;
+    }
+
+    /**
+     * Returns current number of tasks.
+     *
+     * @return Current number of tasks.
+     */
+    public int getLength() {
+        return this.taskList.size();
+    }
+
+    /**
+     * Marks task at index as done.
+     *
+     * @param index Index of task to mark as done.
+     * @throws AntoException Throws AntoException if there is an IO Exception.
+     */
+    public void markTaskAsDone(int index) throws AntoException {
+        String relativePath = "data/anto.txt";
+        Path absolutePath = Paths.get(relativePath).toAbsolutePath();
+        File antoFile = absolutePath.toFile();
+
+        String tempRelativePath = "data/tempFile.txt";
+        Path tempAbsolutePath = Paths.get(tempRelativePath).toAbsolutePath();
+        File tempFile = tempAbsolutePath.toFile();
+
+        try {
+            Scanner sc = new Scanner(antoFile);
+            FileWriter writer = new FileWriter(tempFile);
+
+            int line = 0;
+
+            while (sc.hasNextLine()) {
```

Comment:
Should all this logic be placed in Storage class instead? Since it is handling reading and writing to storage.